test/correctness/bit_counting.cpp

```diff
@@ -6,57 +6,74 @@
 
 using namespace Halide;
 
-uint32_t local_popcount(uint32_t v) {
-    uint32_t count = 0;
+template <typename T>
+T local_popcount(T v) {
+    T count = 0;
     while (v) {
         if (v & 1) ++count;
         v >>= 1;
     }
     return count;
 }
 
-uint32_t local_count_trailing_zeros(uint32_t v) {
-    for (uint32_t b = 0; b < 32; ++b) {
-        if (v & (1 << b))
+template <typename T>
+T local_count_trailing_zeros(T v) {
+    const int bits = sizeof(T)*8;
+    for (T b = 0; b < bits; ++b) {
+        if (v & (1 << b)) {
             // found a set bit
             return b;
+        }
     }
     return 0;
 }
 
-uint32_t local_count_leading_zeros(uint32_t v) {
-    for (uint32_t b = 0; b < 32; ++b) {
-        if (v & (1 << (31 - b)))
+template <typename T>
+T local_count_leading_zeros(T v) {
+    const int bits = sizeof(T)*8;
+    for (T b = 0; b < bits; ++b) {
+        if (v & (1 << (bits - 1 - b))) {
             // found a set bit
             return b;
+        }
     }
     return 0;
 }
 
-std::string as_bits(uint32_t v) {
+template <typename T>
+std::string as_bits(T v) {
+    const int bits = sizeof(T)*8;
     std::string ret;
-    for (int i = 31; i >= 0; --i)
+    for (int i = bits - 1; i >= 0; --i)
         ret += (v & (1 << i)) ? '1' : '0';
     return ret;
 }
 
-int main() {
-    Image<uint32_t> input(256);
+Var x("x");
+
+void schedule(Func f, const Target &t) {
+    // TODO: Add GPU schedule where supported.
+    f.vectorize(x, 16);
+}
+
+template <typename T>
+int test_bit_counting(const Target &target) {
+    const int bits = sizeof(T)*8;
+    Image<T> input(256);
     for (int i = 0; i < 256; i++) {
         if (i < 16)
             input(i) = i;
         else if (i < 32)
-            input(i) = 0xfffffffful - i;
```

Comment:
Why this change? It broke the 32-bit build bots. I'll revert it, but I'm not sure why it happened